### PR TITLE
:sparkles: ERv2 MSK stack

### PR DESCRIPTION
**Pull request by chassing**
This is the initial release of the External Resource Version 2 MSK stack. It implements every feature as we have in `terraform-resources`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
+.PHONY: build
+build:
+	$(CONTAINER_ENGINE) build -t $(IMAGE_NAME):${IMAGE_TAG} $(foreach arg,$(BUILD_ARGS),--build-arg $(arg)) -f dockerfiles/Dockerfile .
 
-build-deploy:
-	$(CONTAINER_ENGINE) build --build-arg MAKE_TARGET=pypi $(foreach arg,$(BUILD_ARGS),--build-arg $(arg)) .
-.PHONY: build-deploy
+.PHONY: push
+push:
+	$(CONTAINER_ENGINE) push ${BASE_IMAGE}:${IMAGE_TAG}
```
Comment (hemslo) [inline Makefile:30]: do we have `login` before `push`?

File: dockerfiles/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+RUN poetry install --quiet --no-root --without dev
+
+# the source code
+ARG CODE_ROOT
+COPY $CODE_ROOT ./$CODE_ROOT
+
+# cdktf files
+COPY cdktf.json entrypoint.sh ./
+
+ENTRYPOINT [ "bash", "-e", "entrypoint.sh" ]
```
Comment (hemslo) [inline dockerfiles/Dockerfile:36]: can the entrypoint point to `entrypoint.sh` directly without `bash -e`?

File: Makefile (hunk 1 above)
Comment (chassing) [reply Makefile:30]: The `docker login` is done via [pre-script in the Jenkins job](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/jenkins/app-sre/external-resources/job-templates.yaml#L37)

File: dockerfiles/Dockerfile (hunk 2 above)
Comment (chassing) [reply dockerfiles/Dockerfile]: Done in [63065f5](https://github.com/app-sre/er-aws-msk/pull/1/commits/63065f53455f6eafd3ff5ab3c5f940f01651aa04)

File: tests/test_stack.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@pytest.fixture
+def synthesized(stack: Stack) -> str:
+    """Fixture to provide the synthesized stack."""
+    return Testing.synth(stack)
+
+
+# No idea why this is failing if executed in the docker container but not locally!?!
+# def test_stack_is_valid_terraform(stack: Stack) -> None:
+#     """Test stack produces valid Terraform code."""
+#     assert Testing.to_be_valid_terraform(Testing.full_synth(stack))  # noqa: ERA001
```
[the remark this answers is not in the document]
Comment (chassing) [reply tests/test_stack.py]: Removed in [f24612c](https://github.com/app-sre/er-aws-msk/pull/1/commits/f24612c962e59e0dfdb807785d8d363ce60457a2)

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**